jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarProcessor.java

```diff
@@ -49,4 +65,86 @@ public static JarType determineJarType(Path jarPath) throws IOException {
     }
     return JarType.STANDARD;
   }
+
+  /**
+   * Explode jar and create three layers for classes, resources and dependencies on container.
+   *
+   * @param jarPath path to jar file.
+   * @param tempDirPath path to temporary jib local directory.
+   * @return list of {@link FileEntriesLayer}.
+   * @throws IOException if I/O error occurs when opening the jar file or if temporary directory
+   *     provided doesn't exist.
+   */
+  public static List<FileEntriesLayer> explodeStandardJar(Path jarPath, Path tempDirPath)
+      throws IOException {
+    Path localExplodedJarRoot = tempDirPath;
+    ZipUtil.unzip(jarPath, localExplodedJarRoot);
+    List<FileEntriesLayer> layers = new ArrayList<>();
+    Predicate<Path> isClassFile = path -> path.getFileName().toString().endsWith(".class");
+    Predicate<Path> isResourceFile = isClassFile.negate();
+
+    // Determine class and resource files in the directory containing jar contents and create
+    // FileEntriesLayer for each type of layer (classes or resources), while maintaining the
+    // file's original project structure.
+    FileEntriesLayer classesLayer =
+        addDirectoryContentsToLayer(
+                FileEntriesLayer.builder(),
+                localExplodedJarRoot,
+                isClassFile,
+                APP_ROOT.resolve(RelativeUnixPath.get("explodedJar")))
+            .setName("classes")
+            .build();
+    FileEntriesLayer resourcesLayer =
+        addDirectoryContentsToLayer(
+                FileEntriesLayer.builder(),
+                localExplodedJarRoot,
+                isResourceFile,
+                APP_ROOT.resolve(RelativeUnixPath.get("explodedJar")))
+            .setName("resources")
+            .build();
+
+    // Get dependencies from Class-Path in the jar's manifest and add a layer with these
+    // dependencies as entries. If Class-Path in the jar's manifest is not present then skip adding
+    // a dependencies layer.
+    JarFile jarFile = new JarFile(jarPath.toFile());
+    String classPath =
+        jarFile.getManifest().getMainAttributes().getValue(Attributes.Name.CLASS_PATH);
+    if (classPath != null) {
+      List<Path> dependencies =
+          Splitter.onPattern("\\s+")
+              .splitToList(classPath.trim())
+              .stream()
+              .map(Paths::get)
+              .collect(Collectors.toList());
+      FileEntriesLayer.Builder dependenciesLayerBuilder =
+          FileEntriesLayer.builder().setName("dependencies");
+      dependencies.forEach(
+          path ->
+              dependenciesLayerBuilder.addEntry(
+                  path, APP_ROOT.resolve(RelativeUnixPath.get("dependencies")).resolve(path)));
+      layers.add(dependenciesLayerBuilder.build());
+    }
+
+    layers.add(resourcesLayer);
+    layers.add(classesLayer);
+    return layers;
+  }
+
+  private static FileEntriesLayer.Builder addDirectoryContentsToLayer(
+      FileEntriesLayer.Builder builder,
```

Comment:
Sorry, looking at this again, maybe this doesn't need to accept a builder. It can create it locally. That said, it can accept a layer name instead? For example,
```java
private static FileEntriesLayer createLayerFromDirectoryContents(String layerName, ...) {
  FileEntriesLayer.Builder builder = FileEntriesLayer.builder().setName(layerName);
  new DirectoryWalker(...)...;
  return builder.build();
}
```
WDYT?